internal/scenes/game/systems/menu.go

```diff
@@ -0,0 +1,173 @@
+package systems
+
+import (
+	"github.com/ebitenui/ebitenui"
+	"github.com/ebitenui/ebitenui/image"
+	"github.com/ebitenui/ebitenui/widget"
+	"github.com/golang/freetype/truetype"
+	"github.com/hajimehoshi/ebiten/v2"
+	"github.com/hajimehoshi/ebiten/v2/inpututil"
+	"github.com/ubootgame/ubootgame/internal/scenes/game/components"
+	"github.com/yohamta/donburi"
+	"github.com/yohamta/donburi/ecs"
+	"golang.org/x/image/font"
+	"golang.org/x/image/font/gofont/goregular"
+	"image/color"
+	"log"
+	"os"
+)
+
+type menusystem struct {
+	UI        *ebitenui.UI
+	menuEntry *donburi.Entry
+}
+
+var Menu = &menusystem{
+	UI: &ebitenui.UI{Container: setupMenu()},
+}
+
+func setupMenu() *widget.Container {
+
+	// construct a new container that serves as the root of the UI hierarchy
+	rootContainer := widget.NewContainer(
+		// the container will use a plain color as its background
+		widget.ContainerOpts.BackgroundImage(image.NewNineSliceColor(color.NRGBA{0, 0, 0, 175})),
+
+		// the container will use an anchor layout to layout its single child widget
+		widget.ContainerOpts.Layout(widget.NewAnchorLayout()),
+	)
+
+	menuContainer := widget.NewContainer(
+		widget.ContainerOpts.WidgetOpts(
+			widget.WidgetOpts.LayoutData(widget.AnchorLayoutData{
+				HorizontalPosition: widget.AnchorLayoutPositionCenter,
+				VerticalPosition:   widget.AnchorLayoutPositionCenter,
+			}),
+		),
+		widget.ContainerOpts.Layout(
+			widget.NewRowLayout(
+				widget.RowLayoutOpts.Direction(widget.DirectionVertical),
+				widget.RowLayoutOpts.Spacing(30),
+			),
+		),
+	)
+
+	newGameButton := createMenuButton("New Game", func(args *widget.ButtonClickedEventArgs) {
```

Comment:
feels bad to pass anonymous functions for the button handlers (but they don't have to be of course)

Reply:
Also wasn't sure how to implement actually restarting the game

Reply:
They can be named methods on the system.
Also, sounds like we can use events for this, like in `events/display_updated.go`